### ISSUE TITLE
Input type="date" polyfill (date picker): Setting "min" as future date causes picker to open up at max date

### ISSUE THREAD

**Issue by WalidBounouar** [wet-boew/wet-boew#8248]
Hi,

I want to set the "min" attribute as a future date, but doing so causes the picker to open at the max date (if not set, opens at year 2100 which I believe is default max). However, the dates are disabled correctly if you navigate to them.

This happens only on some browsers. The native chrome picker is fine. I encounter the issue on Firefox 52.5 and IE 11.

Here’s code when inspecting page and screenshots.
`<input id="frmCCI-completion_date" name="frmCCI-completion_date" min="2018-01-11" size="10" type="date" class="form-control" />`

![image](https://user-images.githubusercontent.com/22263288/34782938-2db96928-f5f8-11e7-96c1-0cae2bd8fc68.png)![image](https://user-images.githubusercontent.com/22263288/34782944-32410e6a-f5f8-11e7-9846-1ab59045ef65.png)

I believe we are using wet v4.0.26 where I am.

I looked in issues and did not find the same problem flagged before. Do you have an idea on what the problem is and how to fix it?

Thanks

**Comment by kodecount** [comment]
I was able to reproduce with latest version of WET on Safari also. When MIN is future dated the calendar opens in the year 2100.

**Comment by kodecount** [comment]
Seem to be related to the follow if statement in the polyfill:

```
if ( today >= state.minDate && today <= state.maxDate ) {
 initDate = today;
} else {
 initDate = state.maxDate;
}
```
Code can be found at [datepicker.js#L120](https://github.com/wet-boew/wet-boew/blob/master/src/polyfills/datepicker/datepicker.js#L120)

**Comment by WalidBounouar** [comment]
Do you have an idea when this could be fixed? Would it be included in a potential v4.0.28 release or could it be added to v4.0.27?

**Comment by duboisp** [comment]
@WalidBounouar we can't provide a time estimate of when this issue would be fixed. 
However, you or someone else can submit a PR with a fix. The submission date of that PR with the necessary time to review it and having the contributor to fix any-reported issue would determinate which version of WET it will be included.

FYI - The version 4.0.27 is already released.

**Comment by kodecount** [comment]
I don't mind submitting the PR. Do you think we can just change it too:
```
if ( today >= state.minDate && today <= state.maxDate ) {
 initDate = today;
} else {
 initDate = state.minDate; ---> FROM MAX TOO MIN?
}
```

**Comment by LaurentGoderre** [comment]
@duboisp @kodecount the calendar polyfill has unit tests. A failing unit test need to be created before the fix.

**Comment by WalidBounouar** [comment]
I am in talks with my superiors to check if I can take time to find a fix and submit a PR. I will keep tabs on this issue in case somebody else. I'm new to this so I would also be unable to provide a time frame.

**Comment by WalidBounouar** [comment]
We decided to use a workaround where I am, so I won't be able to find a fix. I will still follow the progress if this issue.
Thanks

**Comment by duboisp** [comment]
The behavior need to be conform to the HTML 5.3 spec.

* https://www.w3.org/TR/html53/sec-forms.html#date-state-typedate
* https://www.w3.org/TR/html53/sec-forms.html#the-min-and-max-attributes
* https://www.w3.org/TR/html53/rendering.html#the-input-element-as-domain-specific-widgets

I quickly reviewed and I didn't saw a mention about the default behavior when opening with a min date that is in the future.

Next step would be:
* How that use case is implemented in the [browser that support it](https://caniuse.com/#feat=input-datetime)? 
* Is there some UX research about that use case?

**Comment by kodecount** [comment]
FireFox v58: opens the calendar in the current month.
Chrome v68: opens the calendar based on the minimum set.

**Comment by LaurentGoderre** [comment]
The minimum date should be the right behavior.

**Comment by kodecount** [comment]
Any documents on the unit test you want or is that just a working example?

**Comment by LaurentGoderre** [comment]
basically there should be two unit tests for when the minimum date is in the future and when the maximum date is in the past that resembles this one:

https://github.com/wet-boew/wet-boew/blob/master/src/polyfills/datepicker/test.js#L477-L499

**Comment by WalidBounouar** [comment]
Hi,
I would like to know if this issue number is sufficient to track progress on this bug? I don't know if you have another bug tracking system at Treasury Board.

We would like to be able to see when this is fixed so that we can move to a new release of WET.

**Comment by duboisp** [comment]
@WalidBounouar yes this issue number should be sufficient to track progress on this bug. 

If having this issue fixed matter for you or for your organisation, the best option will be to spent your time on resolving it and submitting a PR. If you can work on a fix and need help, contact me and we can have a quick chat. You can find my info on GCconnex, on GCcollab or you can reach me via twitter.

**Comment by WalidBounouar** [comment]
@duboisp thanks for the speedy reply.

If I can get the time to find a fix for this, I'll mention it here. You're contact info is very much appreciated.

**Comment by thekodester** [comment]
I just submitted a PR, but unsure if it need a unit test in the commit.
